packages/react-web/src/inbox/useEnhanceConversation.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  ProfileId,
+  Profile,
+  ReadResult,
+  useLazyProfile,
+  NotFoundError,
+  UnspecifiedError,
+} from '@lens-protocol/react';
+import { CachedConversation } from '@xmtp/react-sdk';
+import { useEffect, useMemo } from 'react';
+
+import { extractPeerProfileId } from './helpers';
+import { EnhancedConversation } from './types';
+
+/**
+ * @experimental
+ */
+export type EnhanceConversationRequest = {
+  /**
+   * Currently authenticated profile
+   */
+  profile: Profile;
+  /**
+   * XMTP conversation to enhance
+   */
+  conversation: CachedConversation;
+};
+
+/**
+ * Enhance XMTP conversation with a profile of the conversation's peer
+ *
+ * @category Inbox
+ * @group Hooks
+ * @experimental
+ *
+ * @param args - {@link EnhanceConversationRequest}
+ *
+ * @example
+ * ```tsx
+ * const { data: enhancedConversation, loading } = useEnhanceConversation({
+ *   conversation,
+ *   profile: session.profile,
+ * });
+ * ```
+ */
+export function useEnhanceConversation({
+  profile,
+  conversation,
+}: EnhanceConversationRequest): ReadResult<EnhancedConversation, NotFoundError | UnspecifiedError> {
+  const peerProfileId = useMemo(
+    (): ProfileId | undefined =>
+      extractPeerProfileId(conversation.context?.conversationId, profile.id),
+    [conversation, profile],
+  );
+
+  const { data: peerProfile, error, loading, execute } = useLazyProfile();
+
+  useEffect(() => {
+    if (peerProfileId) {
+      void execute({ forProfileId: peerProfileId });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
is this because of `execute`?

Reply:
yes, in general there is issue with hook being rendered multiple times, skipping `execute` helped a bit